### PR TITLE
Allow coroutines in service methods

### LINKED ISSUE #24: No non-blocking service methods
Browsing through the examples and docs, it doesn't appear to be possible to have a `ServiceInterface` with non-blocking (`async`) method calls.

This is a bit of an issue, as my [dbus daemon](https://github.com/Appadeia/QueryKit) often does long operations, meaning that all other method calls are delayed until the larger call finishes.

### PR DESCRIPTION
**Pull request by michallowasrzechonek-silvair**
fixes #24

### CODE REVIEW

File: dbus_next/message_bus.py [hunk 1]
```diff
@@ -604,21 +625,21 @@ def _find_message_handler(self, msg):
...
 
-    def _default_ping_handler(self, msg):
-        return Message.new_method_return(msg)
+    def _default_ping_handler(self, msg, send_reply):
+        send_reply(Message.new_method_return(msg))
 
-    def _default_get_machine_id_handler(self, msg):
+    def _default_get_machine_id_handler(self, msg, send_reply):
         if self._machine_id:
             self.send(Message.new_method_return(msg, 's', self._machine_id))
```
Comment (acrisci) [inline dbus_next/message_bus.py]: I think you meant to use `send_reply()` here.

**Comment by acrisci** [comment]
Run `make format` to run `yapf` on the source files.

**Comment by michallowasrzechonek-silvair** [comment]
If this approach looks good, I think it should also work for asynchronous properties, by replacing `getattr` and `setattr` with similar callback-based setters, cf.
 * https://github.com/SilvairGit/python-dbus-next/blob/async_service_methods/dbus_next/message_bus.py#L717-L719 
 * https://github.com/SilvairGit/python-dbus-next/blob/async_service_methods/dbus_next/message_bus.py#L728-L729.

**Comment by acrisci** [comment]
Yeah this approach seems good. That should work for properties, as long as it doesn't break the existing user interface because I like how that works right now.

**Comment by michallowasrzechonek-silvair** [comment]
Alright, I'll try to come up with something that does't change the API. That would be a separate PR, though.

**Comment by acrisci** [comment]
Looks good thanks :+1:

**Pull request merged**